vite.config.mts

```diff
@@ -0,0 +1,142 @@
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+import { optimizeLodashImports } from '@optimize-lodash/rollup-plugin';
+import react from '@vitejs/plugin-react';
+import { PluginOption } from 'vite';
+import svgr from 'vite-plugin-svgr';
+import { analyzer } from 'vite-bundle-analyzer';
+import RailsPlugin from 'vite-plugin-rails';
+import { VitePWA } from 'vite-plugin-pwa';
+
+import { defineConfig, UserConfigFnPromise, UserConfig } from 'vite';
+import postcssPresetEnv from 'postcss-preset-env';
+
+import { MastodonServiceWorkerLocales } from './config/vite/plugin-sw-locales';
+
+const jsRoot = path.resolve(__dirname, 'app/javascript');
+const entrypointRoot = path.resolve(jsRoot, 'entrypoints');
+
+export const config: UserConfigFnPromise = async ({ mode, command }) => {
+  const entrypointFiles = await fs.readdir(entrypointRoot);
+  const entrypoints: Record<string, string> = entrypointFiles.reduce(
+    (acc, file) => {
+      const name = path.basename(file).replace(/\.tsx?$/, '');
+      acc[name] = path.resolve(entrypointRoot, file);
+      return acc;
+    },
+    {} as Record<string, string>,
+  );
+  return {
+    root: jsRoot,
+    css: {
+      postcss: {
+        plugins: [
+          postcssPresetEnv({
+            features: {
+              'logical-properties-and-values': false,
+            },
+          }),
+        ],
+      },
+    },
+    resolve: {
+      alias: {
+        mastodon: path.resolve(jsRoot, 'mastodon'),
```

Comment:
Do we still need this, or does Vite uses the aliases from `tsconfig.json`?

Reply:
Vite does not pull aliases from there, no. there are plugins if we want that.

Reply:
What do you think? I have a feeling that if we can have everything defined in only one place (`tsconfig`) then it's cleaner and will cause less issues in the future if we add or remove aliases?

Reply:
i think it's worth doing, but if it takes more time than expected to just manually copy- we don't have a lot of aliases, after all.